eng/tools/rush-runner.js

```diff
@@ -0,0 +1,112 @@
+const fs = require('fs');
+const path = require('path');
+const process = require('process');
+const { spawn } = require('child_process');
+
+const parseArgs = () => {
+  if (process.argv.length < 3 || process.argv.some(a => ['-h', '--help'].includes(a.toLowerCase()))) {
+    console.error('Usage: rush-runner.js <action> [<servicename>...] [args...]');
+    console.error('Example: rush-runner.js build keyvault storage --verbose');
+    process.exit(1);
+  }
+
+  let inFlags = false;
+  const services = [], flags = [];
+  const [scriptPath, action, ...givenArgs] = process.argv.slice(1);
+  const baseDir = path.resolve(`${path.dirname(scriptPath)}/../..`);
+
+  for (const arg of givenArgs) {
+    if (!inFlags && arg.startsWith('-')) {
+      inFlags = true;
+    }
+
+    if (inFlags) {
+      flags.push(arg);
+    } else {
+      if (arg !== '*') { // exclude special value "*" meaning all libraries
+        services.push(arg);
+      }
+    }
+  }
+  return [baseDir, action, services, flags];
+};
+
+const getPackageJsons = (searchDir) => {
+  return fs.readdirSync(searchDir)
+    .filter(f => !f.startsWith('arm-')) // exclude libraries starting with "arm-"
+    .map(f => path.join(searchDir, f, 'package.json')) // turn potential directory names into package.json paths
+    .filter(f => fs.existsSync(f)); // only keep paths for files that actually exist
+};
+
+const getServicePackages = (baseDir, serviceDirs) => {
+  const packageNames = [], packageDirs = [];
+  for (const serviceDir of serviceDirs) {
+    const searchDir = path.resolve(path.join(baseDir, 'sdk', serviceDir));
+    const packageJsons = getPackageJsons(searchDir);
+    for (const filePath of packageJsons) {
+      const contents = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+      if (contents['sdk-type'] === 'client') {
+        packageNames.push(contents.name);
+        packageDirs.push(path.dirname(filePath));
+      }
+    }
+  }
+
+  return [packageNames, packageDirs];
+};
+
+const spawnNode = (cwd, ...args) => {
+  const y = `\u001B[33m`;
+  const x = `\u001B[39m`;
+
+  console.log(`Executing: "node ${args.join(' ')}" in ${cwd}\n\n`);
+  const proc = spawn('node', args, { cwd });
```

Comment:
TODO: This doesn't appear to block, so need to make multiple calls to `spawn` run serially rather than in parallel.

Reply:
Fixed in e813480